react-hot-button-component/my-app/src/App.tsx

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+import './App.css';
+import HotButton from './HotButton';
+
+function App() {
+  const [count, setCount] = useState(0);
```

Comment:
Since only the HotButton component uses this state, there's not a good reason for putting the state up here, in App. Typically, you want to keep state down at the components that need it, and only move it up if you have to share it among components.